#!/usr/bin/env python
PACKAGE = "limo_follower"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("maxSpeed", double_t, 0, "Maximum speed", 0.5, 0.0, 1.0)
gen.add("targetDist", double_t, 0, "Target distance to keep from object", 0.8, 0.3, 2.0)

# PID参数
gen.add("P_v", double_t, 0, "P value for linear velocity", 0.5, 0.0, 2.0)
gen.add("I_v", double_t, 0, "I value for linear velocity", 0.0, 0.0, 2.0)
gen.add("D_v", double_t, 0, "D value for linear velocity", 0.1, 0.0, 2.0)

gen.add("P_w", double_t, 0, "P value for angular velocity", 1.0, 0.0, 3.0)
gen.add("I_w", double_t, 0, "I value for angular velocity", 0.0, 0.0, 3.0)
gen.add("D_w", double_t, 0, "D value for angular velocity", 0.1, 0.0, 3.0)

# 激光雷达参数
gen.add("winSize", int_t, 0, "Window size for laser scan filtering", 5, 1, 20)
gen.add("deltaDist", double_t, 0, "Delta distance for laser scan filtering", 0.2, 0.01, 1.0)
gen.add("maxTrackingDistance", double_t, 0, "Maximum tracking distance in meters", 5.0, 1.0, 10.0)

# 滤波器参数
gen.add("filter_size", int_t, 0, "Size of velocity filter window", 5, 1, 20)
gen.add("filter_alpha", double_t, 0, "Low-pass filter coefficient (0-1)", 0.3, 0.1, 1.0)

# 死区参数
gen.add("angle_deadzone", double_t, 0, "Angle error deadzone in radians", 0.05, 0.0, 0.5)
gen.add("distance_deadzone", double_t, 0, "Distance error deadzone in meters", 0.05, 0.0, 0.5)

exit(gen.generate(PACKAGE, "limo_follower", "laser_params"))
